src/test/groovy/org/gradle/wrapperupgrade/GradleWrapperUpgradePluginFuncTest.groovy

```diff
@@ -66,6 +66,7 @@ class GradleWrapperUpgradePluginFuncTest extends Specification {
             .withProjectDir(testProjectDir)
             .withPluginClasspath()
             .withGradleVersion(GradleVersion.version('5.6.4').version)
+            .withEnvironment(System.getenv() + ["JAVA_HOME": determineJavaHome()])
```

Comment:
Is `System.getenv()` required?

Reply:
Yes. The `withEnvironment` call replaced the whole env, so `git`, which is required, is not accessible.